tests/clients/test_ngsi_v2_cb.py

```diff
@@ -229,6 +231,147 @@ def test_entity_operations(self):
+            self.assertEqual(client.get_entity(entity_id=self.entity.id),
+                             res_entity)
+
+    @clean_test(fiware_service=settings.FIWARE_SERVICE,
+                fiware_servicepath=settings.FIWARE_SERVICEPATH,
+                cb_url=settings.CB_URL)
+    def test_entity_update(self):
+        """
+        Test different ways (post, update, override, patch) to update entity
+        Both with the update scenario
+        1) append attribute
+        2) update existing attribute value
+        1) delete attribute
+        
+        Returns:
+
+        """
+        with ContextBrokerClient(
+                url=settings.CB_URL,
+                fiware_header=self.fiware_header) as client:
+            entity_init = self.entity.model_copy(deep=True)
+            attr_init = entity_init.get_attribute("temperature")
+            attr_init.metadata = {
+                    "metadata_init": {
+                        "type": "Text",
+                        "value": "something"}
+                }
+            attr_append = NamedContextAttribute(**{
+                "name": 'pressure',
+                "type": 'Number',
+                "value": 1050})
+            entity_init.update_attribute(attrs=[attr_init])
+
+            # Post
+            if "post":
+                client.post_entity(entity=entity_init, update=True)
+                entity_post = entity_init.model_copy(deep=True)
+                # 1) append attribute
+                entity_post.add_attributes(attrs=[attr_append])
+                client.post_entity(entity=entity_post, patch=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+                # 2) update existing attribute value
+                attr_append = NamedContextAttribute(**{
+                    "name": 'pressure',
+                    "type": 'Number',
+                    "value": 2050})
+                entity_post.update_attribute(attrs=[attr_append])
+                client.post_entity(entity=entity_post, patch=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+                # 3) delete attribute
+                entity_post.delete_attributes(attrs=[attr_append])
+                client.post_entity(entity=entity_post, update=True)
+                self.assertEqual(client.get_entity(entity_id=entity_post.id),
+                                 entity_post)
+
+            # update_entity()
+            if "update_entity":
+                client.post_entity(entity=entity_init, update=True)
+                entity_update = entity_init.model_copy(deep=True)
+                # 1) append attribute
+                entity_update.add_attributes(attrs=[attr_append])
+                with self.assertRaises(requests.RequestException):
+                    client.update_entity(entity=entity_update, append_strict=True)
```

Comment:
@djs0109 This call still makes an update to the entity. So we could add the follwing assertion:

```
self.assertEqual(client.get_entity(entity_id=entity_update.id),
                                 entity_update)
```

Reply:
You are right, I thought that the request will be completely blocked